www/js/diary/list.js

```diff
@@ -315,17 +345,35 @@ angular.module('emission.main.diary.list', ['ui-leaflet',
       $scope.$apply(function () {
         $scope.data = Timeline.data;
         $scope.datepickerObject.inputDate = Timeline.data.currDay.toDate();
-        $scope.data.currDayTrips.forEach(function (trip, index, array) {
-          addUnpushedMode(trip);
-          addUnpushedPurpose(trip);
-          PostTripManualMarker.addUnpushedIncidents(trip);
-        });
-        $scope.data.currDayTripWrappers = Timeline.data.currDayTrips.map(
-          DiaryHelper.directiveForTrip);
-        // $scope.data.currAnalysedDayTripWrappers = $scope.data.currDayTripWrappers.filter(function(el){
-        //   return el.data.id.indexOf('unprocessed_') === -1;
-        // });
-        $ionicScrollDelegate.scrollTop(true);
+        
+        function routineTasks () {
+          $scope.data.currDayTrips.forEach(function (trip, index, array) {
+            addUnpushedMode(trip);
+            addUnpushedPurpose(trip);
+            PostTripManualMarker.addUnpushedIncidents(trip);
+          });
+          $scope.data.currDayTripWrappers = Timeline.data.currDayTrips.map(
+            DiaryHelper.directiveForTrip);
+          // $scope.data.currAnalysedDayTripWrappers = $scope.data.currDayTripWrappers.filter(function(el){
+          //   return el.data.id.indexOf('unprocessed_') === -1;
+          // });
+          $ionicScrollDelegate.scrollTop(true);
+        }
+        if ($scope.unifiedResults === null) {
+          var tq = { key: 'write_ts', startTs: 0, endTs: moment().endOf('day').unix(), };
+          Promise.all([
+            UnifiedDataLoader.getUnifiedMessagesForInterval(MODE_CONFIRM_KEY, tq),
```

Comment:
The `UnifiedDataLoader` loads data from both the local cache and the server 
https://github.com/e-mission/e-mission-phone/blob/master/www/js/services.js#L233

so falling back to the `UnifiedDataLoader` if the local call fails (e.g. `getLocalTripMode().then....`) makes an extra, unnecessary call to the local database. You should be able to just use the `unifiedResults` directly.

Reply:
This is for compatibility with the existing phone that is running.

Reply:
Not sure what you mean by this. You are currently merging to `rk-unsw` and you are the only ones using that branch. So compatibility is not an immediate issue.

wrt to other branches, the PR that you got the original change from is unmerged, so nobody else is using it either.

Given that this is essentially the first implementation of this change, I would rather that we did it right wrt performance instead of prioritizing compatibility with an implementation that nobody is using.

Also, at least from your code, the right way doesn't seem to be that much harder than your way, and will make the final implementation significantly easier to understand. Please let me know if you disagree and what the concrete issues with using the right way are.